scheduler/plugin/resultstore/store.go

```diff
@@ -233,5 +235,11 @@ func (s *Store) applyWeightOnScore(pluginName string, score int64) int64 {
 }
 
 func (s *Store) DeleteData(k key) {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	s.deleteData(k)
+}
+
+func (s *Store) deleteData(k key) {
```

Comment:
nits: let's add the note about the lock

```suggestion
// deleteData deletes the result stored with the given key.
// Note: we assume the store lock is already acquired.
func (s *Store) deleteData(k key) {
```